appshell/appshell_extensions_mac.mm

```diff
@@ -1175,6 +1175,154 @@ int32 SetMenuTitle(CefRefPtr<CefBrowser> browser, ExtensionString command, Exten
+            
+            NSString* bundlePath = [[NSBundle mainBundle] bundlePath];
+            NSString* sourcePath;
+            NSRange range = [bundlePath rangeOfString: @"/Frameworks/"];
+            
+            if (range.location == NSNotFound) {
+                sourcePath = [[NSBundle mainBundle] pathForResource: @"Brackets" ofType: @"sh"];
+            } else {
+                sourcePath = [bundlePath substringToIndex:range.location];
+                sourcePath = [sourcePath stringByAppendingString:@"/Resources/Brackets.sh"];
+            }
+
+            std::string sourceFile = [sourcePath UTF8String];
+
+            std::vector<std::string> args;
+
+            // Now execute all these steps one by one
+            //  1. Remove existing symlink
+            //  2. Check for existence of the directory and create one if required.
+            //  3  Create symlink at /usr/local/bin.
+            
+            // *** Removing the existing symlink ***
+            
+            args.push_back(rmArgs);
+            args.push_back(destFile);
+            
+            toolStatus = _RunToolWithAdminPrivileges(authorizationRef, rmTool, args);
+            
+            if( toolStatus == errAuthorizationSuccess) {
+                
+                // *** Check and create if the directory is not present ***
+                
+                args.clear();
+                
+                args.push_back(mkDirArgs);
+                args.push_back(destFolder);
+                
+                toolStatus = _RunToolWithAdminPrivileges(authorizationRef, mkDirTool, args);
+                
+                if( toolStatus == errAuthorizationSuccess) {
+                    
+                    // *** Go ahead and create the symlink now ***
+                    
+                    args.clear();
+                    
+                    args.push_back(lnArgs);
+                    args.push_back(sourceFile);
+                    args.push_back(destFile);
+                    
+                    toolStatus = _RunToolWithAdminPrivileges(authorizationRef, lnTool, args);
+
+                    if( toolStatus != errAuthorizationSuccess)
+                        errorCode = ERR_CL_TOOLS_SYMLINKFAILED;
+                    
+                }
+                else {
+                    errorCode = ERR_CL_TOOLS_MKDIRFAILED;
+                }
+            }
+            else{
+                if(toolStatus == errAuthorizationCanceled)
+                    errorCode = ERR_CL_TOOLS_CANCELLED;
+                else
+                    errorCode = ERR_CL_TOOLS_RMFAILED;
+            }
+
+        } catch (...) {
+            // This is empty as the below statements will take care of
+            // releasing authorizationRef.
+            errorCode = ERR_CL_TOOLS_SERVFAILED;
+        }
+        
+        if(authorizationRef) {
+            AuthorizationFree(authorizationRef, kAuthorizationFlagDestroyRights);
```

Comment:
I don't think any extension developer can reuse this. The required operation has succesfully been executed. I think flagging this as an error to the user is wrong as the script is already installed at `/usr/local/bin`. We could however track this internally.